fix: add horizontal scroller on middleware table to prevent overflow on the small screen

Mohit5Upadhyay opened the pull request:
<!--
Thank you for your pull request. Please provide a description and 
note the Certificate of Origin below. 

-->

## Description
This PR resolve issue: #1967 there middleware guide contains a `table` with an `image and callouts` that exceeds the viewport width on mobile devices, causing horizontal overflow and breaking layout.

## Solve 
- Wrapped the diagram table in a horizontally scrollable div (.mw-fig-scroll) with overflow-x: auto  based on media-query

https://github.com/user-attachments/assets/bbc407d0-1ea5-4aeb-af0e-37af2c191df2



This preserves the readability of the middleware diagram without distorting the layout on mobile.

Closes #1967 

<!--
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

Mohit5Upadhyay committed:
fix(docs): add horizontal scroll wrapper to middleware diagram to prevent overflow on mobile (#1967)

Mohit5Upadhyay commented:
Hi team 👋

This PR resolves #1967 , where the middleware guide diagram was overflowing the mobile viewport and causing layout issues.
I’ve wrapped the table containing the diagram in a horizontally scrollable container. 

Let me know if you’d prefer any adjustments or some tweaks..
Thanks!

ShubhamOulkar commented on en/guide/writing-middleware.md:
```diff
@@ -24,6 +24,7 @@ If the current middleware function does not end the request-response cycle, it m
 
 The following figure shows the elements of a middleware function call:
 
+<div class="mw-fig-scroll">
```
```suggestion
<div class="table-scroller">
```

ShubhamOulkar reviewed (changes requested):
All the changes are irrelevant. Just use predefined CSS class `table-scroller`.

I just checked it looks good. 
<img width="722" height="430" alt="image" src="https://github.com/user-attachments/assets/923607cf-966e-42cb-b6f1-c3b87a8f8c9c" />

Mohit5Upadhyay committed:
fix(docs): replace mw-fig-scroll with table-scroller as per review feedback (#1967)

Mohit5Upadhyay commented:
Hi @ShubhamOulkar,

I’ve updated the markup to use the `.table-scroller` class as suggested and removed the custom CSS wrapper.
> All the changes are irrelevant. Just use predefined CSS class table-scroller.

 Thanks!

ShubhamOulkar committed:
apply scroller width for same cross browser behaviour

ShubhamOulkar reviewed (approved):
LGTM
